### PR TITLE
Working end-to-end

### PR DESCRIPTION
**Pull request by WesRoach**
Tested by running these commands in this order and verified all is well.

```
make run-dflag
make clean
make run
make clean
make test
make clean
make debug
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -69,7 +86,7 @@ debug: build-debug
...
 clean:
 	@ find ./bin/ ! -name '.gitignore' -type f -exec rm -f {} +
-	@ find tests/workdir/output ! -name '.gitignore' -type f -exec rm -f {} +
-	@ find tests/workdir/temp ! -name '.gitignore' -type f -exec rm -f {} +
-	@ find workdir/output ! -name '.gitignore' -type f -exec rm -f {} +
-	@ find workdir/temp ! -name '.gitignore' -type f -exec rm -f {} +
+	@ rm -rf workdir/temp
+	@ rm -rf workdir/output
+	@ rm -rf tests/workdir/temp
+	@ rm -rf tests/workdir/output
```
Comment (WesRoach) [inline Makefile:92]: FileManager manages creating the directories if they don't exist, and we've removed the `.gitignores` in a previous commit for the same reason. Simplified this a bit.

File: src/Map.cpp [hunk 2]
```diff
@@ -77,7 +77,7 @@ std::vector<std::string> Map::tokenize(std::string line) {
 void Map::exportData(std::string word, int numLines, int lineNum) {
     buffer += "(" + word + ", 1)\n";
     if ((buffer.size() >= bufferSize) || (lineNum == numLines - 1)) {
-        std::string tmpFile = boost::filesystem::path(inputFilename).stem().string() + "Output.txt";
+        std::string tmpFile = boost::filesystem::path(inputFilename).stem().string() + ".txt";
```
Comment (WesRoach) [inline src/Map.cpp:80]: @josephcranston `Map` is the only component writing to the temporary directory. Removed the `Output` addition which makes keep tracking of the file being processed a bit easier when dumping things into the output directory. Updated the test runner to match.

You might have had other, potentially future plans for this and would like it to stay - lmk and I can resolve the issue another way. This seemed like the easiest approach.

File: src/Reduce.cpp [hunk 3]
```diff
@@ -17,11 +24,9 @@ void Reduce::reduce(const std::string &key, const std::vector<int> &values) {
...
-    std::cout << "Writing (" + key + ", " << value << ") to " + this->_output_dir << std::endl;
-
-    // Write SUCCESS file to output directory
-    // Should use File Management Class
-    std::cout << "Writing SUCCESS file to " + this->_output_dir << std::endl;
+    // Write result to output file using fileManager
+    fileManager->appendToFile(
+        this->fileManager->getOutputDirectory(),
+        this->outputFilename,
+        "(" + key + ", " + std::to_string(value) + ")\n");
```
Comment (WesRoach) [inline src/Reduce.cpp:31]: I can add a buffer in future PR if we want better performance.

File: src/Map.cpp (hunk 2 above)
Comment (josephcranston) [reply src/Map.cpp:80]: @WesRoach No problem - I only added the "Output" as a placeholder.

File: Makefile (hunk 1 above)
Comment (squeeve) [reply Makefile:92]: LGTM

File: src/Reduce.cpp (hunk 3 above)
Comment (squeeve) [reply src/Reduce.cpp:31]: I'll resolve this and merge for now, but this is something we can definitely talk about in tonight's meeting.

**Pull request merged**